WordPressFlux/WordPressFlux/Sources/QueryStore.swift

```diff
@@ -116,59 +120,50 @@ open class QueryStore<State, Query>: StatefulStore<State>, Unsubscribable {
 
 }
 
-// In ideal world, those wouldn't be extensions on `Encodable`/`Codable`, but rather refinements of
-// `QueryStore` in which `Store` conforms to `Codable` — then we would just change behavior _inside_ the `QueryStore`,
-// based on whether the type conforms to `Codable` or not.
-// Regretfully, Swift type system (or my Swift skills) does't allow to express that —
-// you can't just do `let codableSelf = self as? QueryStore<Codable, Query>` because
-// _Codable itself doesn't conform to Codable_.
-private extension Encodable {
-    func storeCachedState(at url: URL) {
-        do {
-            let jsonEncoder = JSONEncoder.init()
-            let encodedStore = try jsonEncoder.encode(self)
-
-            try encodedStore.write(to: url, options: [.atomic])
-        } catch {
-            NSLog("[PluginStore Error] \(error)")
-        }
+public extension Encodable {
+    func saveJSON(at url: URL, using encoder: JSONEncoder = JSONEncoder()) throws {
+        let encodedStore = try encoder.encode(self)
+        try encodedStore.write(to: url, options: [.atomic])
     }
 }
 
-private extension Decodable {
-    static func cachedState(from cachedURL: URL) -> Self? {
+public extension Decodable {
```

Comment:
Sorry to change my mind and nitpick, but after discussing NSLog I realize this is inside the flux framework, so we might want to keep these two helpers `internal` (and in a separate file)